src/streamlink/plugins/filmon.py

```diff
@@ -95,7 +95,7 @@ class Filmon(Plugin):
         )
     """, re.VERBOSE)
 
-    _channel_id_re = re.compile(r'channel_id\s*?=\s*"(\d+)"')
+    _channel_id_re = re.compile(r"""channel_id\s*?=\s*"?(\d+)["']""")
```

Comment:
You have `"` at the start of the id, but `"` or `'` at the end, should it be `["']` on both sides?

Reply:
No it's not nessecary, the id itself isn't quoted in that case.

Reply:
What @beardypig meant was that usually when you're matching (optional) strings, you should try to match the same kind of quotation mark at the end of the string which was used at the beginning of it. The regex should look something like that:
```python
r"""channel_id\s*=\s*(?P<quote>['"]?)(?P<value>\d+)(?P=quote)"""
```
This also replaces the unnecessary `\s*?` with a simple `\s*` and gives each capture group a name, as it increments the position of the value capture group by one. This needs to get fixed in the code below...